### PR TITLE
Warning about blacklisted members in application

### LINKED ISSUE #1742: Warning to handler when an applicant is blacklisted

### PR DESCRIPTION
**Pull request by opqdonut**
fixes #1742

# Definition of Done / Review checklist

## Reviewability
- [x] link to issue
- [x] consider adding screenshots for ease of review

## API
- [x] API is documented and shows up in Swagger UI
- [x] API is backwards compatible or completely new

## Documentation
- ~update changelog if necessary~
- [x] components are added to guide page

## Testing
- [x] complex logic is unit tested
- [x] valuable features are integration / browser / acceptance tested automatically

## Follow-up
- ~new tasks are created for pending or remaining tasks~
- [x] no critical TODOs left to implement

### CODE REVIEW

**Comment by opqdonut** [comment]
Screenshot:
blacklist-warning

**Comment by opqdonut** [comment]
I'll follow up with some refactorings I noticed along the way

File: src/clj/rems/application/model.clj [hunk 1]
```diff
@@ -541,6 +542,18 @@
...
              {}))))
 
+(defn- get-blacklisted-users [application blacklisted?]
+  (let [all-members (application-util/applicant-and-members application)
+        all-resources (map :resource/ext-id (:application/resources application))]
+    (vec
+     (for [member all-members
+           resource all-resources
+           :when (blacklisted? (:userid member) resource)]
+       {:blacklist/user member :blacklist/resource {:resource/ext-id resource}}))))
```
Comment (Macroz) [inline src/clj/rems/application/model.clj]: I would expect this function to group by user. Perhaps do that or rename the function to something like `get-blacklisted-user-resources`. The name does not imply that it returns the resources as well or that it returns one row per user-resource -pair.

File: src/cljs/rems/application.cljs [hunk 2]
```diff
@@ -62,6 +62,16 @@
...
 
+(defn- blacklist-warning [application]
+  (let [blacklist (:application/blacklisted-users application)]
+    (when (not (empty? blacklist))
+      [:div.alert.alert-danger
+       (text :t.form/alert-blacklisted-users)
+       (into [:ul]
+             (for [entry blacklist]
+               [:li (get-member-name (:blacklist/user entry))
+                ": " (get-in entry [:blacklist/resource :resource/ext-id])]))])))
```
Comment (Macroz) [inline src/cljs/rems/application.cljs]: I think from the UX perspective it doesn't make sense to list here the resource ids but we should instead use the catalogue items that are present in the application and their names.

File: src/clj/rems/application/model.clj (hunk 1 above)
Comment (opqdonut) [reply src/clj/rems/application/model.clj]: changed the function & field name to `blacklist`

File: src/cljs/rems/application.cljs (hunk 2 above)
Comment (opqdonut) [reply src/cljs/rems/application.cljs]: did this in the frontend

**Comment by opqdonut** [comment]
Now the blacklist shows the catalogue item. I opted not to render the urn / more info links for clarity.

blacklist

**Comment by opqdonut** [comment]
Hmm the same resource can point to multiple catalogue items, I'll need to fix that

**Pull request merged**